src/org/sosy_lab/java_smt/solvers/mathsat5/Mathsat5AbstractProver.java

```diff
@@ -97,18 +97,32 @@ private long buildConfig(Set<ProverOptions> opts) {
   /** add needed options into the given map. */
   protected abstract void createConfig(Map<String, String> pConfig);
 
+  private <T> T exec(Callable<T> closure) throws SolverException, InterruptedException {
+    long hook = context.addTerminationTest(curEnv);
+    T value = null;
+    try {
+      value = closure.call();
+    } catch (Throwable t) {
+      Throwables.propagateIfPossible(t, IllegalStateException.class, SolverException.class);
+      Throwables.propagateIfPossible(t, InterruptedException.class);
```

Comment:
There is a bug here: This will silently swallow all checked exceptions except for the mentioned classes. One always has to make sure to end this list of `propagateIfPossible` with a catch all that throws any remaining exception (e.g., by wrapping it in an `AssertionError`).

Furthermore: `IllegalStateException` is unchecked and is thus redundant here. The JavaDoc of `propagateIfPossible` says that this method is discouraged and mentions a replacement.

But of course, as argued above, all this could go away anyway.